libraries/helpers.rb

```diff
@@ -307,5 +307,25 @@ def package_options
 
       options
     end
+
+    #
+    # Checks the deprecated properties of chef-ingredient and prints warning
+    # messages if any of them are being used.
+    #
+    def check_deprecated_properties
```

Comment:
I think it would be a good idea to add some validation against `PRODUCT_MATRIX`. We can add that with a separate PR.

With the above function is everything being handled? 

* 'push-client' => 'push-jobs-client',
* 'push-server' => 'push-jobs-server'
* chef-server-ha-provisioning